### PR TITLE
Enhance/8130 audience setup for secondary users

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue:

- #8130

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Comment by kuasha420** [comment]
The failing VRT test (`google-site-kit_Modules_Analytics4_Widgets_DashboardOverallPageMetricsWidgetGA4_LoadedEntityURL_0_document_0_small.png`) is unrelated and will be fixed in #9101

**Review by techanvil — CHANGES_REQUESTED** [review]
Hi @kuasha420, nice work so far, it's a complicated beast this one. However, a few changes are needed. Most critically, the widget is not actually appearing on the view-only dashboard at present.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationErrorWidget.js [hunk 1]
```diff
@@ -102,6 +103,9 @@ function AudienceSegmentationErrorWidget( { Widget, errors } ) {
 								buttonVariant="danger"
 								getHelpClassName="googlesitekit-error-retry-text"
 								onRetry={ handleRetry }
+								overrideShowRetryButton={
+									typeof onRetry === 'function'
+								}
```
[the remark this answers is not in the document]
Comment (kuasha420) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationErrorWidget.js]: Thanks @techanvil. This all makes sense, but not using `ReportErrorActions` has the drawback of the "Retry didn't work? Get help" part not rendering. Not sure if that is an issue. 

I have made the changes you suggested in this [commit](https://github.com/google/site-kit-wp/pull/9219/commits/86cefd377486ea44fd7337e3c891516eee168247). Let me know if this is preferred, or we can revert and tackle this in a future issue. 

Cheers.

[the remark this answers is not in the document]
Comment (techanvil) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationErrorWidget.js]: Thanks @kuasha420. As per the [design](https://www.figma.com/design/7pSrkEy8t00BcYRAi9LjjH/Audience-Segmentation?node-id=2258-36640&node-type=FRAME&m=dev) and [current implementation](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/modules-analytics4-components-audiencesegmentation-dashboard-audiencesegmentationerrorwidget--default) of the error tile, we don't actually need to show "Retry didn't work? Get help".

However, we _do_ need to show the **Request access** button for the insufficient permissions variant.

That being the case, I think we can keep the code as committed but add the following change in `SecondaryUserSetupWidget` to ensure the **Retry** button is only shown when it's _not_ an insufficient permission error:

```js
showRetryButton={ ! isInsufficientPermissionsError( setupError ) }
```

image

image

**Review by techanvil — APPROVED** [review]
Thanks @kuasha420! Nice work. Although, while testing this during CR, I came across a few, hopefully final issues.

- The audience sync was being triggered from the `AudienceTile` while the initial setup was still in progress and this resulted in the `audienceSegmentationSetupComplete` being overridden from `false` to an empty string.
- The `AudienceSegmentationSetupCTAWidget` was still appearing for secondary users even though the primary admin had set the feature up.
- There were a couple of tests that were proving flaky in CI.

I pushed fixes for all of these issues to save another round trip.

With that all said and done, this LGTM!

**Pull request merged**